backend/backend/common/room.py

```diff
@@ -28,17 +34,45 @@ def room_id(self):
     def add_member(self, member: RoomMember):
         self._members.update({member.id: member})
 
-    def update_leader(self, player_id, new_leader_id):
+    def update_leader(self, player_id: str, new_leader_id: str):
         player = self._members.get(player_id)
         new_leader = self._members.get(new_leader_id)
 
         if not player or not new_leader:
-            raise PlayerNotInRoom(players=[player_id, new_leader_id])
+            raise PlayerNotInRoomError(players=[player_id, new_leader_id])
 
         if not player.leader:
-            raise PlayerIsNotLeader(player_name=player.name)
+            raise PlayerIsNotLeaderError(player_name=player.name)
 
         player.leader = False
         new_leader.leader = True
 
+    def get_room_member(self, player_id: str) -> RoomMember:
+        member = self._members.get(player_id)
+        if not member:
+            raise PlayerNotInRoomError(players=[player_id])
+
+        return member
+
+    async def _broadcast(self, message: BaseModel):
+        message = Message(timestamp=datetime.now(),
+                          content=message)
+        for member in self._members.values():
+            await member.websocket.send(message)
+
+    async def _on_receive(self, member: RoomMember, payload: Any):
+        new_content = None
+        # if self.turn.current_song and payload.content == self.turn.current_song:
+        #     score = self.turn.calculate_score(payload)
+        #     new_content = member.add_to_score(score)
+
+        message = new_content or ClientContent(sender=member.name,
+                                               text=payload)
```

Comment:
what's the purpose of `new_content`? it would always be None...

Reply:
welp, this is mb- will uncomment this code segment- but this will make the callback fail becuase we didn't  design the tarn system yer to it fails on turn.current_song